src/components/ApiPublishModal/ApiPublishModal.tsx

```diff
@@ -119,12 +140,25 @@ const ApiDeployModal: React.FC = () => {
         setActiveStep(activeStep + 1);
       }
 
+      if (!apiContent?.openapi) {
+        return;
+      }
+
+      // api info
+      if (activeStep === 1) {
+        const {name, namespace} = values;
+
+        dispatch(setNewApiContent({name, namespace: namespace || 'default', openapi: apiContent.openapi}));
+
+        setActiveStep(activeStep + 1);
+      }
+
       if (!apiContent) {
         return;
       }
 
       // validation
-      if (activeStep === 1) {
+      if (activeStep === 2) {
```

Comment:
I'm wondering if something like the below would help with the readability of steps:
```
const orderedSteps = ['someStep', 'otherStep', 'publishStep'] as const;
type StepType = typeof list[number];

const [activeStep, setActiveStep] = useState<StepType>();

if (activeStep === 'someStep') {
   // Do ...
}

const activeStepIndex = useMemo(() => orderedSteps.indexOf(activeStep), [activeStep]);
```